pyqt-apps/siriushla/si_ap_fofb/custom_widgets.py

```diff
@@ -968,6 +975,210 @@ def _setupErrCntTab(self):
+        lay_fru.addWidget(head_lbl, 0, 0, 2, 1)
+
+        sr_lbl = QLabel(
+            '<h4>Soft Reset</h4>', self, alignment=Qt.AlignCenter)
+        lay_fru.addWidget(sr_lbl, 0, 1, 2, 1)
+
+        row = 2
+
+        for i in range(1, 21):
+            pv_name = pv_list_fru['SoftRst']
+
+            pbt = QPushButton(qta.icon('fa5s.ellipsis-h'), '', self)
+            pbt.setObjectName('pbt')
+            pbt.setStyleSheet(
+                '#pbt{min-width:25px; max-width:25px; icon-size:20px;}')
+            pbt.setDefault(False)
+            pbt.setAutoDefault(False)
+            Window = create_window_from_widget(
+                AFCAdvancedSettings, title='AFC Advanced Settings'
+            )
+            util.connect_window(
+                pbt, Window, parent=self, prefix=self.prefix, area_index=i
+            )
+
+            lbl = QLabel(f"IA-{i:02d}RaBPM", self, alignment=Qt.AlignCenter)
+            lbl.setStyleSheet(
+                'QLabel{border: none; background: transparent;}'
+                )
+            lbl.setObjectName('lbl_afcname')
+            hwid = QWidget()
+            hwid.setObjectName('wid')
+            hwid.setStyleSheet('#wid{border: 1px solid gray;}')
+            hlay = QHBoxLayout(hwid)
+            hlay.setContentsMargins(2, 0, 2, 0)
+            hlay.addWidget(pbt)
+            hlay.addWidget(lbl)
+            lay_fru.addWidget(hwid, row, 0)
+
+            cmd_button = SiriusPushButton(
+                self, label='Reset', icon=qta.icon('mdi.restart'),
+                init_channel=f"IA-{i:02d}RaBPM:{afcs_pvs.DIS}-AMC-{2}:{pv_name}"
+                )
+            cmd_button.setObjectName('cmd_button')
+            cmd_button.setStyleSheet(
+                '#cmd_button{min-width:120px; max-width:120px; icon-size:20px;}')
+
+            lay_fru.addWidget(cmd_button, row, 1)
+
+            pv_name = pv_list_fru['SoftRstSts']
+            label_mon = SiriusLabel(
+                self, f"IA-{i:02d}RaBPM:{afcs_pvs.DIS}-AMC-{2}:{pv_name}",
+                alignment=Qt.AlignCenter)
+            label_mon.setStyleSheet(
+                'QLabel{border: none; background: transparent;}'
+                )
+
+            lay_fru.addWidget(label_mon, row, 2, alignment=Qt.AlignLeft)
+
+            row += 1
+
+        return self._build_scroll_area(wid)
+
+    def _build_scroll_area(self, widget):
+        area = QScrollArea(self)
+        area.setSizeAdjustPolicy(QScrollArea.AdjustToContentsOnFirstShow)
+        area.setWidgetResizable(True)
+        area.setWidget(widget)
+        widget.setObjectName('widget')
+        widget.setStyleSheet(
+            '#widget{background-color: transparent;}'
+            'QLabel{border: 1px solid gray; min-height: 1.5em;}'
+            '#lbl_bpmname{border: 0px solid gray; min-height: 1.5em;}'
+            '#led_status{border: 1px solid gray; min-height: 1.5em;}')
+        return area
+
+class AFCAdvancedSettings(BaseWidget):
```

Comment:
This class seems to be very similar to the one defined for BPMs or for AFCTiming, it seems to be possible to have a generic class, that can be define maybe in subpackage `as_ap_rabpmmon`, and can receive the board version (3.1 or 4.0) and create a generic GUI for all the device types.